include/boost/histogram/python/accumulators/weighted_sum.hpp

```diff
@@ -0,0 +1,120 @@
+//  * A special constructor added for construction from Python
+
+#pragma once
+
+#include <boost/core/nvp.hpp>
+#include <boost/histogram/fwd.hpp> // for weighted_sum<>
+#include <type_traits>
+
+namespace boost {
+namespace histogram {
+namespace python {
+
+/// Holds sum of weights and its variance estimate
+template <typename RealType>
+struct weighted_sum {
+  public:
+    weighted_sum() = default;
+    explicit weighted_sum(const RealType &value) noexcept
+        : sum_of_weights_(value)
+        , sum_of_weights_squared_(value) {}
+    weighted_sum(const RealType &value, const RealType &variance) noexcept
+        : sum_of_weights_(value)
+        , sum_of_weights_squared_(variance) {}
+
+    /// Increment by one.
+    weighted_sum &operator++() { return operator+=(1); }
+
+    /// Increment by value.
+    template <typename T>
+    weighted_sum &operator+=(const T &value) {
+        sum_of_weights_ += value;
+        sum_of_weights_squared_ += value * value;
+        return *this;
+    }
+
+    /// Added another weighted sum.
+    template <typename T>
+    weighted_sum &operator+=(const weighted_sum<T> &rhs) {
+        sum_of_weights_ += static_cast<RealType>(rhs.sum_of_weights_);
+        sum_of_weights_squared_ += static_cast<RealType>(rhs.sum_of_weights_squared_);
+        return *this;
+    }
+
+    /// Scale by value.
+    weighted_sum &operator*=(const RealType &x) {
+        sum_of_weights_ *= x;
+        sum_of_weights_squared_ *= x * x;
+        return *this;
+    }
+
+    bool operator==(const RealType &rhs) const noexcept {
+        return sum_of_weights_ == rhs && sum_of_weights_squared_ == rhs;
+    }
+
+    template <typename T>
+    bool operator==(const weighted_sum<T> &rhs) const noexcept {
+        return sum_of_weights_ == rhs.sum_of_weights_
+               && sum_of_weights_squared_ == rhs.sum_of_weights_squared_;
+    }
+
+    template <typename T>
+    bool operator!=(const T &rhs) const noexcept {
+        return !operator==(rhs);
+    }
+
+    /// Return value of the sum.
+    const RealType &value() const noexcept { return sum_of_weights_; }
+
+    /// Return estimated variance of the sum.
+    const RealType &variance() const noexcept { return sum_of_weights_squared_; }
+
+    // lossy conversion must be explicit
+    template <class T>
+    explicit operator T() const {
+        return static_cast<T>(sum_of_weights_);
+    }
+
+    template <class Archive>
+    void serialize(Archive &ar, unsigned /* version */) {
+        ar &make_nvp("sum_of_weights", sum_of_weights_);
+        ar &make_nvp("sum_of_weights_squared", sum_of_weights_squared_);
+    }
+
+    RealType sum_of_weights_         = RealType();
+    RealType sum_of_weights_squared_ = RealType();
+};
+
+} // namespace python
+} // namespace histogram
+} // namespace boost
+
+#ifndef BOOST_HISTOGRAM_DOXYGEN_INVOKED
+namespace std {
+template <class T, class U>
+struct common_type<boost::histogram::accumulators::weighted_sum<T>,
```

Comment:
to clarify, strictly speaking these are wrong now, because you want common_type specializations for your versions of weighted_sum, which these here are common_type specializations for the original boost::histogram accumulators. but never mind all this, just remove them :)